packages/vite-plugin-svelte/src/preprocess.ts

```diff
@@ -76,7 +77,14 @@ function viteStyle(config: InlineConfig | ResolvedConfig = {}): {
 		const { code, map } = await transform(content, moduleId);
 
 		mapSourcesToRelative(map, moduleId);
-
+		if (map?.file === moduleId) {
+			map.file = filename;
+		}
+		if (map?.sources?.map) {
```

Comment:
wouldn't this have to be called before mapSourcesToRelative (and then use filename instead of moduleId as second arg for that)